app/main.py

```diff
@@ -2,24 +2,68 @@
 
 
 class IntegerRange:
-    pass
+    """Descriptor data validation"""
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.name = name
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{value} must be in range:"       # noqa
+                             f" {self.min_amount} - {self.max_amount}")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self,
+                 name: str,
+                 age: int,
+                 weight: int,
+                 height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
```

Comment:
This is a design issue. The `Visitor` class has no restrictions on the values of `age`, `weight`, and `height`. However, in your `SlideLimitationValidator` subclasses, you are imposing restrictions on these attributes. This could lead to inconsistencies or errors. You might want to consider using your `IntegerRange` descriptor in the `Visitor` class to ensure that the values of `age`, `weight`, and `height` are always within acceptable ranges.